frontend/mr-admin-flate/src/components/avtaler/AvtalePersonvernForm.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  Alert,
+  BodyShort,
+  Checkbox,
+  GuidePanel,
+  HGrid,
+  HStack,
+  Label,
+  Radio,
+  VStack,
+} from "@navikt/ds-react";
+import { useFormContext } from "react-hook-form";
+import { InferredAvtaleSchema } from "../redaksjonelt-innhold/AvtaleSchema";
+import styles from "./AvtalePersonvernForm.module.scss";
+import { useTiltakstype } from "@/api/tiltakstyper/useTiltakstype";
+import { personopplysningToTekst } from "@/utils/Utils";
+import { ControlledRadioGroup } from "../skjema/ControlledRadioGroup";
+import { useEffect } from "react";
+import { addOrRemove } from "mulighetsrommet-frontend-common/utils/utils";
+
+interface Props {
+  tiltakstypeId?: string;
+}
+
+export function AvtalePersonvernForm({ tiltakstypeId }: Props) {
+  const { register, setValue, watch } = useFormContext<InferredAvtaleSchema>();
+  const { data: tiltakstype } = useTiltakstype(tiltakstypeId);
+
+  const watchPersonopplysninger = watch("personopplysninger");
+  useEffect(() => {
+    if (watchPersonopplysninger.length === 0 && tiltakstype) {
+      setValue("personopplysninger", tiltakstype.personopplysninger.ALLTID);
+    }
+  }, [watchPersonopplysninger, tiltakstype]);
+
+  if (!tiltakstypeId) {
+    return (
+      <div className={styles.container}>
+        <Alert variant="info">Tiltakstype må velges før personvern kan redigeres.</Alert>
+      </div>
+    );
+  }
+
+  return (
+    <VStack gap="4" className={styles.container}>
+      <GuidePanel poster className={styles.guide_panel}>
+        Huk av de personopplysningene som er avtalt i databehandleravtalen. NAV tiltaksenhet/fylke
+        er ansvarlig for at listen er i samsvar med gjeldende databehandleravtale.
+      </GuidePanel>
+      <HGrid columns={2}>
+        <VStack>
+          <Label size="small">Opplysninger om brukeren som alltid kan/må behandles</Label>
+          <BodyShort size="small" textColor="subtle">
+            Fjern avhukingen hvis noen av opplysningene ikke er relevante for denne avtalen.
+          </BodyShort>
+          {tiltakstype?.personopplysninger?.ALLTID.map((p) => (
+            <Checkbox
+              checked={watchPersonopplysninger.includes(p)}
+              onChange={() =>
+                setValue("personopplysninger", addOrRemove(watchPersonopplysninger, p))
+              }
+              size="small"
+              key={p}
+            >
+              {personopplysningToTekst(p)}
+            </Checkbox>
+          ))}
+        </VStack>
+        <VStack justify="space-between">
+          <VStack>
+            <Label size="small">
+              Opplysninger om brukeren som ofte er nødvendig og relevant å behandle
+            </Label>
+            <BodyShort size="small" textColor="subtle">
+              Huk av for de opplysningene som er avtalt i databehandleravtalen.
+            </BodyShort>
```

Comment:
Kan dra denne ut til en egen komponent i bunn av denne filen og så gjenbruke den siden den brukes likt to steder

Reply:
Når jeg tenker meg om kan du sikkert lage en mer generisk komponent av hele `VStack`-koden (label, bodyshort, personopplysninger.<FREKVENS>.map osv...`. Da tror jeg hele denne filen blir enda enklere.